driver-core/src/main/com/mongodb/internal/connection/SaslAuthenticator.java

```diff
@@ -121,6 +126,16 @@ private void throwIfSaslClientIsNull(final SaslClient saslClient) {
         }
     }
 
+    private void verifySaslClientComplete(final SaslClient saslClient, final SingleResultCallback<Void> callback) {
+        if (saslClient.isComplete()) {
```

Comment:
The change I suggested for the sync version, if acceptable, should also be applied here (and tested in the same way).

Reply:
Done